livelog/livelog.go

```diff
@@ -97,13 +103,25 @@ func (l *LiveLog) connectInputStream() error {
 	}
 	client := new(http.Client)
 	go func() {
-		// TODO: this is a HORRENDOUS hack - need to fix this Basically we need
-		// to wait until put port is opened which is some time after the
+		// We need to wait until put port is opened which is some time after the
 		// livelog process has started...
-		time.Sleep(500 * time.Millisecond)
+		waitForPortToBeActive(60022)
 		// since we waited so long, maybe livelog service isn't running now, so
 		// ignore any error and response we get back...
 		client.Do(req)
 	}()
 	return nil
 }
+
+func waitForPortToBeActive(port int) {
+	deadline := time.Now().Add(60 * time.Second)
+	for time.Now().Before(deadline) {
+		conn, err := net.DialTimeout("tcp", "localhost:"+strconv.Itoa(port), 60*time.Second)
```

Comment:
I wrote some code to test the behaviour, and discovered that was how it worked. It started a go routine which slept a few seconds and then started an http server, and in another go routine, it ran the DialTimeout call, which exited immediately. Then I adapted it to look as above, and that worked. I didn't commit the test, just because tests that sleep didn't seem like a good idea, but I guess I could have committed it with short timeouts (like 50ms or so). In the end though, no doubt we'll replace this with TaskCluster worker so maybe not worth the effort of writing extensive tests for this.